endpoints/openrtb2/video_auction_test.go

```diff
@@ -363,6 +363,119 @@ func TestVideoEndpointValidationsPodErrors(t *testing.T) {
+	assert.Equal(t, 2, len(bidRespVideo.AdPods[0].Targeting), "AdPod Targeting length should be 2")
+	assert.Equal(t, "17.00_123_30s", bidRespVideo.AdPods[0].Targeting[0].Hb_pb_cat_dur, "AdPod Targeting first element hb_pb_cat_dur should be 17.00_123_30s")
+	assert.Equal(t, "17.00_456_30s", bidRespVideo.AdPods[0].Targeting[1].Hb_pb_cat_dur, "AdPod Targeting first element hb_pb_cat_dur should be 17.00_456_30s")
+}
+
+func TestVideoBuildVideoResponseMissedCacheForAllBids(t *testing.T) {
+	openRtbBidResp := openrtb.BidResponse{}
+	podErrors := make([]PodError, 0)
+
+	seatBids := make([]openrtb.SeatBid, 0)
+	seatBid := openrtb.SeatBid{}
+
+	bids := make([]openrtb.Bid, 0)
+	bid1 := openrtb.Bid{}
+	bid2 := openrtb.Bid{}
+	bid3 := openrtb.Bid{}
+
+	extBid1 := []byte(`{"prebid":{"targeting":{"hb_bidder":"appnexus","hb_pb":"17.00","hb_pb_cat_dur":"17.00_123_30s","hb_size":"1x1"}}}`)
+	extBid2 := []byte(`{"prebid":{"targeting":{"hb_bidder":"appnexus","hb_pb":"17.00","hb_pb_cat_dur":"17.00_456_30s","hb_size":"1x1"}}}`)
+	extBid3 := []byte(`{"prebid":{"targeting":{"hb_bidder":"appnexus","hb_pb":"17.00","hb_pb_cat_dur":"17.00_406_30s","hb_size":"1x1"}}}`)
+
+	bid1.Ext = extBid1
+	bids = append(bids, bid1)
+
+	bid2.Ext = extBid2
+	bids = append(bids, bid2)
+
+	bid3.Ext = extBid3
+	bids = append(bids, bid3)
+
+	seatBid.Bid = bids
+	seatBids = append(seatBids, seatBid)
+	openRtbBidResp.SeatBid = seatBids
+
+	bidRespVideo, err := buildVideoResponse(&openRtbBidResp, podErrors)
+	assert.Equal(t, true, bidRespVideo == nil, "bid response should be nil")
+	assert.Equal(t, "caching failed for all bids", err.Error(), "error should be caching failed for all bids")
+}
+
+func TestVideoBuildVideoResponsePodErrors(t *testing.T) {
+	openRtbBidResp := openrtb.BidResponse{}
+	podErrors := make([]PodError, 0, 2)
+
+	seatBids := make([]openrtb.SeatBid, 0)
+	seatBid := openrtb.SeatBid{}
+
+	bids := make([]openrtb.Bid, 0)
+	bid1 := openrtb.Bid{}
+	bid2 := openrtb.Bid{}
+
+	extBid1 := []byte(`{"prebid":{"targeting":{"hb_bidder":"appnexus","hb_pb":"17.00","hb_pb_cat_dur":"17.00_123_30s","hb_size":"1x1","hb_uuid":"837ea3b7-5598-4958-8c45-8e9ef2bf7cc1"}}}`)
+	extBid2 := []byte(`{"prebid":{"targeting":{"hb_bidder":"appnexus","hb_pb":"17.00","hb_pb_cat_dur":"17.00_456_30s","hb_size":"1x1","hb_uuid":"837ea3b7-5598-4958-8c45-8e9ef2bf7cc1"}}}`)
+
+	bid1.Ext = extBid1
+	bids = append(bids, bid1)
+
+	bid2.Ext = extBid2
+	bids = append(bids, bid2)
+
+	seatBid.Bid = bids
+	seatBids = append(seatBids, seatBid)
+	openRtbBidResp.SeatBid = seatBids
+
+	podErr1 := PodError{}
+	podErr1.PodId = 222
+	podErr1.PodIndex = 1
+	podErrors = append(podErrors, podErr1)
+
+	podErr2 := PodError{}
+	podErr2.PodId = 333
+	podErr2.PodIndex = 2
+	podErrors = append(podErrors, podErr2)
+
+	bidRespVideo, err := buildVideoResponse(&openRtbBidResp, podErrors)
+	assert.Equal(t, nil, err, "Error shopuld be nil")
+	assert.Equal(t, 3, len(bidRespVideo.AdPods), "AdPods length should be 3")
```

Comment:
This is more of a question: In this particular test case I see that you're sending 2 bids and both of them has `hb_uuid` targeting key set so then in the `buildVideoResponse` function, the if condition on line 369 should be true for both the bids and no adPods should be created, correct? But here we are expecting 3 adPods and I am likely missing something here. Would you mind explaining to me why is that so that I can understand this better?

Reply:
Oh I see what you mean! In this test case the idea is to test only pod level errors, not cache errors. Bid data should be correct, content doesn't matter. 
 There are also 2 more test cases to test cache errors: `TestVideoBuildVideoResponseMissedCacheForAllBids` - no cache for all bids and no pod erros
`TestVideoBuildVideoResponseMissedCacheForOneBid` - just one bid cache error and no pods errors
In `buildVideoResponse` function we need to handle both cache and pod errors to build proper response. I just try to test it with different combinations. 

To be more clear about purpose of `TestVideoBuildVideoResponsePodErrors` test case, I added one more assert:
`assert.Len(t, bidRespVideo.AdPods[0].Targeting, 2, "First ad pod should be correct and contain 2 targeting elements")`
Hope this explanation makes sense for you.